aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -847,14 +849,33 @@ TensorIterator TensorIterator::binary_float_op(Tensor& out, const Tensor& a,
 
 TensorIterator TensorIterator::comparison_op(Tensor& out, const Tensor& a,
     const Tensor& b) {
-  return TensorIteratorConfig()
+  // Note [special-case bool outputs]
```

Comment:
Logic to avoid an unnecessary copy in the functional version of all comparison/logically ops (thanks @mruberry)

We avoid calling `cast_common_dtype_to_outputs()` when the output tensor has `bool` dtype, which would require copying the output tensor.